connectors/sources/outlook.py

```diff
@@ -0,0 +1,916 @@
+    "importance",
+    "categories",
+    "body",
+    "has_attachments",
+    "attachments",
+]
+CONTACT_FIELDS = [
+    "email_addresses",
+    "phone_numbers",
+    "last_modified_time",
+    "display_name",
+    "company_name",
+    "birthday",
+]
+TASK_FIELDS = [
+    "last_modified_time",
+    "due_date",
+    "complete_date",
+    "subject",
+    "status",
+    "owner",
+    "start_date",
+    "text_body",
+    "companies",
+    "categories",
+    "importance",
+    "has_attachments",
+    "attachments",
+]
+CALENDAR_FIELDS = [
+    "required_attendees",
+    "type",
+    "recurrence",
+    "last_modified_time",
+    "subject",
+    "start",
+    "end",
+    "location",
+    "organizer",
+    "body",
+    "has_attachments",
+    "attachments",
+]
+
+END_SIGNAL = "FINISHED"
+
+
+def ews_format_to_datetime(datetime, timezone):
+    """Change datetime format to user account timezone
+    Args:
+        datetime: Datetime in UTC format
+        timezone: User account timezone
+    Returns:
+        Datetime: Date format as user account timezone
+    """
+    if isinstance(datetime, exchangelib.ewsdatetime.EWSDateTime):
+        return (datetime.astimezone(pytz.timezone(str(timezone)))).strftime(
+            "%Y-%m-%dT%H:%M:%SZ"
+        )
+    elif isinstance(datetime, exchangelib.ewsdatetime.EWSDate) or isinstance(
+        datetime, date
+    ):
+        return datetime.strftime("%Y-%m-%d")
+    else:
+        return datetime
+
+
+class TokenFetchFailed(Exception):
+    """Exception class to notify that connector was unable to fetch authentication token from Graph API"""
+
+    pass
+
+
+class UsersFetchFailed(Exception):
+    """Exception class to notify that connector was unable to fetch users from Active Directory"""
+
+    pass
+
+
+class UnauthorizedException(Exception):
+    """Exception class unauthorized calls"""
+
+    pass
+
+
+class RootCAAdapter(requests.adapters.HTTPAdapter):
+    """Class to verify SSL Certificate for Exchange Servers"""
+
+    def __init__(self):
+        self.ssl_verify = False
+        self.ssl_ctx = None
+
+    def cert_verify(self, conn, url, ssl_certificate_file, cert):
+        super().cert_verify(
+            conn=conn, url=url, verify=self.ssl_verify, cert=self.ssl_ctx
+        )
+
+
+class ExchangeUsers:
+    """Fetch users from Exchange Active Directory"""
+
+    def __init__(self, ad_server, domain, exchange_server, user, password, ssl_enabled):
+        self.ad_server = ad_server
+        self.domain = domain
+        self.exchange_server = exchange_server
+        self.user = user
+        self.password = password
+        self.ssl_enabled = ssl_enabled
+
+    def _create_connection(self, server):
+        return Connection(
+            server=server,
+            user=self.user,
+            password=self.password,
+            client_strategy=SAFE_SYNC,
+            auto_bind=True,  # pyright: ignore
+        )
+
+    async def get_users(self):
+        warnings.filterwarnings("ignore")
```

Comment:
Do we really need to disable the warnings here? 

As a side effect we will disable warning for the whole framework (I think) because we never revert back to default level:
```
# Revert back to the default warning behavior
warnings.filterwarnings("default")
```

But if possible I would get rid of this statement and address the root cause of warnings.

Reply:
Warning logs are there in case the SSL is disabled, so we're ignoring that warnings. If the ssl is enabled, warnings won't be there (for SSL). Other than that i am not sure which warnings are there so let's keep a "default" behaviour.